client/my-sites/plugins/plugin-management-v2/plugin-action-status/render-status-message.tsx

```diff
@@ -0,0 +1,180 @@
+						'Deactivated on %(count)s site',
+						'Deactivated on %(count)s sites',
+						translationArgs
+				  ),
+			[ PLUGIN_INSTALLATION_ERROR ]: selectedSite
+				? translate( 'Failed to deactivate' )
+				: failedStatusMessage(
+						translate(
+							'Failed to deactivate on %(count)s site',
+							'Failed to deactivate on %(count)s sites',
+							translationArgs
+						)
+				  ),
+		},
+		[ ENABLE_AUTOUPDATE_PLUGIN ]: {
+			[ PLUGIN_INSTALLATION_IN_PROGRESS ]: selectedSite
+				? translate( 'Enabling auto-updates' )
+				: translate(
+						'Enabling auto-updates on %(count)s site',
+						'Enabling auto-updates on %(count)s sites',
+						translationArgs
+				  ),
+			[ PLUGIN_INSTALLATION_COMPLETED ]: selectedSite
+				? translate( 'Auto-update enabled' )
+				: translate(
+						'Auto-update enabled on %(count)s site',
+						'Auto-update enabled on %(count)s sites',
+						translationArgs
+				  ),
+			[ PLUGIN_INSTALLATION_ERROR ]: selectedSite
+				? translate( 'Failed to enable auto-updates' )
+				: failedStatusMessage(
+						translate(
+							'Failed to enable auto-updates on %(count)s site',
+							'Failed to enable auto-updates on %(count)s sites',
+							translationArgs
+						)
+				  ),
+		},
+		[ DISABLE_AUTOUPDATE_PLUGIN ]: {
+			[ PLUGIN_INSTALLATION_IN_PROGRESS ]: selectedSite
+				? translate( 'Disabling auto-updates' )
+				: translate(
+						'Disabling auto-updates on %(count)s site',
+						'Disabling auto-updates on %(count)s sites',
+						translationArgs
+				  ),
+			[ PLUGIN_INSTALLATION_COMPLETED ]: selectedSite
+				? translate( 'Auto-update disabled' )
+				: translate(
+						'Auto-update disabled on %(count)s site',
+						'Auto-update disabled on %(count)s sites',
+						translationArgs
+				  ),
+			[ PLUGIN_INSTALLATION_ERROR ]: selectedSite
+				? translate( 'Failed to disable auto-updates' )
+				: failedStatusMessage(
+						translate(
+							'Failed to disable auto-updates on %(count)s site',
+							'Failed to disable auto-updates on %(count)s sites',
+							translationArgs
+						)
+				  ),
+		},
+		[ REMOVE_PLUGIN ]: {
+			[ PLUGIN_INSTALLATION_IN_PROGRESS ]: selectedSite
+				? translate( 'Removing Plugin' )
+				: translate(
+						'Removing plugin from %(count)s site',
+						'Removing plugin from %(count)s sites',
+						translationArgs
+				  ),
+			[ PLUGIN_INSTALLATION_ERROR ]: selectedSite
+				? translate( 'Failed to remove plugin' )
+				: translate(
+						'Failed to remove plugin from %(count)s site',
+						'Failed to remove plugin from %(count)s sites',
+						translationArgs
+				  ),
+		},
+	};
+};
+
+export default function RenderStatusMessage( {
+	selectedSite,
+	groupKey,
+	groupedStatues,
+	hasOneStatus,
+}: Props ): ReactElement | null {
+	const currentGroupStatuses = groupedStatues[ groupKey ];
+
+	const statusMessage = getStatusMessage(
+		currentGroupStatuses.length,
+		hasOneStatus,
+		selectedSite
+	)?.[ currentGroupStatuses[ 0 ].action ]?.[ groupKey ];
```

Comment:
I found this one difficult to understand as well. I'd recommend extracting this code to a separate function and documenting it using Docblock.